thunder/core/jit_ext.py

```diff
@@ -1203,6 +1205,46 @@ def propagate_constraints(ctx, inputs, intermediates, computation_trace):
                     front.append(variableify(inp))
 
 
+# lift shape computational logic into top level trace
+# Maybe: replace usage of shape NumberProxy with prim?
+def lift_shape_logic(computation_trace):
```

Comment:
Note for myself. I don't think I needed this for now since we already have `prims.shape` I'll remove this and see how CI looks like.